x-pack/plugins/maps/public/actions/map_actions.ts

```diff
@@ -369,8 +370,12 @@ export function addNewFeatureToIndex(geometry: Geometry | Position[]) {
     }
 
     try {
-      await (layer as IVectorLayer).addFeature(geometry);
+      dispatch(updateEditShape(DRAW_SHAPE.WAIT));
+      await asyncForEach(geometries, async (geometry) => {
+        await (layer as IVectorLayer).addFeature(geometry);
+      });
       await dispatch(syncDataForLayerDueToDrawing(layer));
+      dispatch(updateEditShape(DRAW_SHAPE.SIMPLE_SELECT));
```

Comment:
Can you move `dispatch(updateEditShape(DRAW_SHAPE.SIMPLE_SELECT));` to after the try/catch. If addFeature fails, then edit mode stays in WAIT.